packages/contracts/src/contracts/current/protocol/Exchange/MixinExchangeCore.sol

```diff
@@ -113,10 +116,16 @@ contract MixinExchangeCore is
             require(order.takerAssetAmount > 0);
             require(isValidSignature(orderHash, order.makerAddress, signature));
         }
+        
+        // Validate sender
+        if (order.senderAddress != address(0)) {
+            require(order.senderAddress == msg.sender);
+        }
```

Comment:
So this conditional enforces that there are two valid orders that can be filled directly by the taker:
- One with `senderAddress` set to `NULL_ADDRESS`
- One with `senderAddress` set to `msg.sender`

?

If so, do we want to constrain this further?

Reply:
Hmm, I'm not sure what you mean. What other constraints were you thinking about?

Reply:
I mean should we enforce that if the user is calling `fillOrder` directly that `senderAddress` **must** be `msg.sender`. This would involve first checking if `getCurrentContextAddress` returns `msg.sender`, and if so, enforce that `senderAddress` is too.

Thoughts?